core/src/main/java/org/elasticsearch/index/query/FilteredQueryBuilder.java

```diff
@@ -72,20 +96,64 @@ public FilteredQueryBuilder queryName(String queryName) {
         return this;
     }
 
+    /** Returns the name of this query. */
+    public String queryName() {
+        return queryName;
+    }
+
     @Override
-    protected void doXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject(NAME);
-        if (queryBuilder != null) {
-            builder.field("query");
-            queryBuilder.toXContent(builder, params);
+    protected boolean doEquals(FilteredQueryBuilder other) {
+        return Objects.equals(queryBuilder, other.queryBuilder) &&
+                Objects.equals(filterBuilder, other.filterBuilder) &&
+                Objects.equals(boost, other.boost) &&
+                Objects.equals(queryName, other.queryName);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(queryBuilder, filterBuilder, boost, queryName);
+    }
+
+    @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
```

Comment:
it wouldn't be empty here at this point, it needs to validate the inner query and filter, see #11889